dhcpv4/nclient4/client.go

```diff
@@ -482,7 +501,9 @@ func (c *Client) RequestFromOffer(ctx context.Context, offer *dhcpv4.DHCPv4, mod
 		return nil, fmt.Errorf("unable to create a request: %w", err)
 	}
 
-	response, err := c.SendAndRead(ctx, c.serverAddr, request, IsMessageType(dhcpv4.MessageTypeAck, dhcpv4.MessageTypeNak))
+	response, err := c.SendAndRead(ctx, c.serverAddr, request, IsAll(
+		IsCorrectServer(offer.ServerIdentifier),
```

Comment:
offer.ServerIdentifier()?

Reply:
I was actually just looking for a mention in the RFC for validating this field. It doesn't require it. I'd add a comment here saying something like

// Servers are supposed to only respond to Requests containing their server identifier, but sometimes non-compliant servers respond anyway. Clients are not required to validate this field, but servers are required to include the server identifier in their Offer per RFC 2131 Section 4.3.1 Table 3.

Reply:
Yeah, makes sense :) Unfortunately I know at least one misbehaving server :-)